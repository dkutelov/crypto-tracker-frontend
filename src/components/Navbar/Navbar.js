import { useContext } from 'react';
import { NavLink, useHistory } from 'react-router-dom';

import logo from '../../assets/icons/logo.svg';
import styles from './Navbar.module.css';
import UserContext from '../../context/userContext';

const Navbar = () => {
  const {
    state: { isAuthenticated },
    dispatch,
  } = useContext(UserContext);
  const history = useHistory();

  function handleLogout() {
    dispatch({ type: 'LOGOUT_USER' });
    history.push('/');
  }
  return (
    <header>
      <div className={styles.container}>
        <div>
          <NavLink to='/' className={styles.logoWrapper}>
            <img src={logo} alt='crypto helper' className={styles.logo} />
            <h1 className={styles.heading}>CryptoHelper</h1>
          </NavLink>
        </div>
        <nav>
          <ul className={styles.navWrapper}>
            {isAuthenticated ? (
              <>
                <li>
                  <NavLink to='/portfolio' className={styles.navLink}>
                    My Portfio
                  </NavLink>
                </li>
                <li>
                  <NavLink to='/profile' className={styles.navLink}>
                    Profile
                  </NavLink>
                </li>
                <li className={styles.authBtn} onClick={handleLogout}>
                  Logout
                </li>
              </>
            ) : (
              <li>
                <NavLink to='/auth/login' className={styles.authBtn}>
                  Login
                </NavLink>
              </li>
            )}
          </ul>
        </nav>
      </div>
    </header>
  );
};

export default Navbar;
