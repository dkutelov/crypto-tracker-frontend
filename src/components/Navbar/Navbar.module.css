.container {
  width: 80%;
  max-width: 1400px;
  margin: 0 auto;
  height: 80px;
  display: flex;
  justify-content: space-between;
}

.logoWrapper {
  display: flex;
  align-items: center;
  text-decoration: none;
}

.logo {
  width: 50px;
  height: 50px;
}

.heading {
  padding-left: 10px;
  color: black;
}

.navWrapper {
  list-style: none;
  display: flex;
  align-items: center;
}

.authBtn {
  padding: 7px 14px;
  border-radius: 5px;
  background-color: #fbe300;
  text-decoration: none;
  text-transform: uppercase;
  color: #222222;
  cursor: pointer;
  transition: all 0.3s;
}

.authBtn:hover {
  background-color: #ffd400;
  color: black;
}

.navLink {
  text-decoration: none;
  color: #222222;
  padding: 0 1em;
}
