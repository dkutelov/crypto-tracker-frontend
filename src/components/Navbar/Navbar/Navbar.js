import { useContext } from 'react';
import { NavLink, useHistory } from 'react-router-dom';

import logo from '../../../assets/icons/logo.svg';
import styles from './Navbar.module.css';
import UserContext from '../../../context/userContext';

const Navbar = () => {
  const {
    state: { isAuthenticated },
    dispatch,
  } = useContext(UserContext);
  const history = useHistory();

  function handleLogout() {
    dispatch({ type: 'LOGOUT_USER' });
    history.push('/');
  }
  return (
    <header className={styles.header}>
      <div className={styles.container}>
        <div>
          <NavLink to='/' className={styles.logoWrapper}>
            <img src={logo} alt='crypto helper' className={styles.logo} />
            <h1 className={styles.heading}>Crypterio</h1>
          </NavLink>
        </div>
        <nav className={styles.navigation}>
          <ul className={styles.navWrapper}>
            {isAuthenticated ? (
              <>
                <li className={styles.linkWrapper}>
                  <NavLink to='/portfolio' className={styles.navLink}>
                    Portfio
                  </NavLink>
                </li>
                <li className={styles.linkWrapper}>
                  <NavLink to='/transactions' className={styles.navLink}>
                    Transactions
                  </NavLink>
                </li>
                <li className={styles.linkWrapper}>
                  <NavLink to='/profile' className={styles.navLink}>
                    Profile
                  </NavLink>
                </li>
                <li className={styles.linkWrapper}>
                  <NavLink to='/buy' className={styles.navLinkButton}>
                    Buy
                  </NavLink>
                </li>
                <li className={styles.authBtn} onClick={handleLogout}>
                  Logout
                </li>
              </>
            ) : (
              <li className={styles.linkWrapper}>
                <NavLink to='/auth/login' className={styles.authBtn}>
                  Login
                </NavLink>
              </li>
            )}
          </ul>
        </nav>
      </div>
    </header>
  );
};

export default Navbar;
