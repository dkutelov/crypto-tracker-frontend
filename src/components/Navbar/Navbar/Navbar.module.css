@media only screen and (max-width: 768px) {
  .header {
    display: none;
  }
}

.container {
  width: 80%;
  max-width: 1400px;
  margin: 0 auto;
  height: 80px;
  display: flex;
  justify-content: space-between;
}

.logoWrapper {
  display: flex;
  align-items: center;
  text-decoration: none;
}

.logo {
  width: 50px;
  height: 50px;
}

.heading {
  padding-left: 10px;
  color: black;
}

.navigation {
  display: flex;
  align-items: center;
}

.navWrapper {
  margin: 0;
  list-style: none;
  display: flex;
  align-items: center;
}

.authBtn,
.navLinkButton {
  padding: 7px 14px;
  border-radius: 5px;
  background-color: #fbe300;
  text-decoration: none;
  text-transform: uppercase;
  color: #222222;
  cursor: pointer;
  transition: all 0.3s;
  margin: 0;
}

.authBtn:hover {
  background-color: #ffd400;
  color: black;
}

.navLink {
  text-decoration: none;
  color: #222222;
  padding: 0 1em;
}

.navLinkButton {
  background-color: #27be94;
  color: white;
  margin: 0 0.5em;
}

.navLinkButton:hover {
  background-color: #58e7bd;
}

.linkWrapper {
  margin: 0;
}
