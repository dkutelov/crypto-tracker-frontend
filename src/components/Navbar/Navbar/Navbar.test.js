import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import UserContext from '../../../context/userContext';

import Navbar from './Navbar';

describe('Navbar Component', () => {
  const state = {
    isAuthenticated: false,
  };
  it('Should match snapshot for non auth users', () => {
    const component = renderer.create(
      <UserContext.Provider value={{ state }}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </UserContext.Provider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  state.isAuthenticated = true;
  it('Should match snapshot for auth users', () => {
    const component = renderer.create(
      <UserContext.Provider value={{ state }}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </UserContext.Provider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
