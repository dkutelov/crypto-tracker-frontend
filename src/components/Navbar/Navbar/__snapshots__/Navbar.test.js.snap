// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component Should match snapshot for auth users 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div>
      <a
        aria-current="page"
        className="logoWrapper active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="crypto helper"
          className="logo"
          src="logo.svg"
        />
        <h1
          className="heading"
        >
          Crypterio
        </h1>
      </a>
    </div>
    <nav
      className="navigation"
    >
      <ul
        className="navWrapper"
      >
        <li
          className="linkWrapper"
        >
          <a
            aria-current={null}
            className="navLink"
            href="/portfolio"
            onClick={[Function]}
          >
            Portfio
          </a>
        </li>
        <li
          className="linkWrapper"
        >
          <a
            aria-current={null}
            className="navLink"
            href="/transactions"
            onClick={[Function]}
          >
            Transactions
          </a>
        </li>
        <li
          className="linkWrapper"
        >
          <a
            aria-current={null}
            className="navLink"
            href="/profile"
            onClick={[Function]}
          >
            Profile
          </a>
        </li>
        <li
          className="linkWrapper"
        >
          <a
            aria-current={null}
            className="navLinkButton"
            href="/buy"
            onClick={[Function]}
          >
            Buy
          </a>
        </li>
        <li
          className="authBtn"
          onClick={[Function]}
        >
          Logout
        </li>
      </ul>
    </nav>
  </div>
</header>
`;

exports[`Navbar Component Should match snapshot for non auth users 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div>
      <a
        aria-current="page"
        className="logoWrapper active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="crypto helper"
          className="logo"
          src="logo.svg"
        />
        <h1
          className="heading"
        >
          Crypterio
        </h1>
      </a>
    </div>
    <nav
      className="navigation"
    >
      <ul
        className="navWrapper"
      >
        <li
          className="linkWrapper"
        >
          <a
            aria-current={null}
            className="navLink"
            href="/portfolio"
            onClick={[Function]}
          >
            Portfio
          </a>
        </li>
        <li
          className="linkWrapper"
        >
          <a
            aria-current={null}
            className="navLink"
            href="/transactions"
            onClick={[Function]}
          >
            Transactions
          </a>
        </li>
        <li
          className="linkWrapper"
        >
          <a
            aria-current={null}
            className="navLink"
            href="/profile"
            onClick={[Function]}
          >
            Profile
          </a>
        </li>
        <li
          className="linkWrapper"
        >
          <a
            aria-current={null}
            className="navLinkButton"
            href="/buy"
            onClick={[Function]}
          >
            Buy
          </a>
        </li>
        <li
          className="authBtn"
          onClick={[Function]}
        >
          Logout
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
